controllers/istiocni/istiocni_controller.go

```diff
@@ -141,32 +142,37 @@ func (r *Reconciler) installHelmChart(ctx context.Context, cni *v1.IstioCNI) err
 		BlockOwnerDeletion: ptr.Of(true),
 	}
 
+	versionName, err := istioversions.ResolveVersionName(cni.Spec.Version)
+	if err != nil {
+		return fmt.Errorf("failed to apply profile: %w", err)
+	}
+
 	// get userValues from Istio.spec.values
 	userValues := cni.Spec.Values
 
 	// apply image digests from configuration, if not already set by user
-	userValues = applyImageDigests(cni, userValues, config.Config)
+	userValues = applyImageDigests(versionName, userValues, config.Config)
 
 	// apply userValues on top of defaultValues from profiles
 	mergedHelmValues, err := istiovalues.ApplyProfilesAndPlatform(
-		r.Config.ResourceDirectory, cni.Spec.Version, r.Config.Platform, r.Config.DefaultProfile, cni.Spec.Profile, helm.FromValues(userValues))
+		r.Config.ResourceDirectory, versionName, r.Config.Platform, r.Config.DefaultProfile, cni.Spec.Profile, helm.FromValues(userValues))
 	if err != nil {
 		return fmt.Errorf("failed to apply profile: %w", err)
 	}
 
-	_, err = r.ChartManager.UpgradeOrInstallChart(ctx, r.getChartDir(cni), mergedHelmValues, cni.Spec.Namespace, cniReleaseName, ownerReference)
+	_, err = r.ChartManager.UpgradeOrInstallChart(ctx, r.getChartDir(versionName), mergedHelmValues, cni.Spec.Namespace, cniReleaseName, ownerReference)
 	if err != nil {
 		return fmt.Errorf("failed to install/update Helm chart %q: %w", cniChartName, err)
 	}
 	return nil
 }
 
-func (r *Reconciler) getChartDir(cni *v1.IstioCNI) string {
-	return path.Join(r.Config.ResourceDirectory, cni.Spec.Version, "charts", cniChartName)
+func (r *Reconciler) getChartDir(versionName string) string {
+	return path.Join(r.Config.ResourceDirectory, versionName, "charts", cniChartName)
 }
```

Comment:
Now that this function no longer takes the IstioCNI object, it can be consolidated with the other `getChartDir()` functions in the other controllers.

Reply:
Agreed. Currently all the controllers are not sharing a common base. I tried to avoid expending this PR with too much refactoring.